From d7758b208c3c0e747b55801c00ce61eb4caf9cac Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Thu, 1 Nov 2018 16:15:43 +0200
Subject: [PATCH 459/706] enetc: Address txbdr_free() todo - free dma mappings
 and skbs

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
(cherry picked from commit 3fd0ca3414e92ac0a10ba88bd768657ffb2df892)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 drivers/net/ethernet/freescale/enetc/enetc.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/enetc/enetc.c b/drivers/net/ethernet/freescale/enetc/enetc.c
index 37f0de751e54..7e5869811c52 100644
--- a/drivers/net/ethernet/freescale/enetc/enetc.c
+++ b/drivers/net/ethernet/freescale/enetc/enetc.c
@@ -753,7 +753,10 @@ static int enetc_alloc_txbdr(struct enetc_bdr *txr)
 
 static void enetc_free_txbdr(struct enetc_bdr *txr)
 {
-	int size;
+	int size, i;
+
+	for (i = 0; i < txr->bd_count; i++)
+		enetc_unmap_tx_buff(txr, &txr->tx_swbd[i]);
 
 	size = txr->bd_count * sizeof(union enetc_tx_bd);
 
@@ -762,7 +765,6 @@ static void enetc_free_txbdr(struct enetc_bdr *txr)
 
 	vfree(txr->tx_swbd);
 	txr->tx_swbd = 0;
-	// TODO: free tx_ring dma mappings and skbs
 }
 
 static int enetc_alloc_tx_resources(struct enetc_ndev_priv *priv)
-- 
2.17.1

